kubernetes/config/kube_config.c

```diff
@@ -21,15 +24,34 @@ static int setBasePath(char **pBasePath, char *basePath)
     return -1;
 }
 
-static char *kubeconfig_mk_cert_key_tempfile(const char *b64data)
+static int is_cert_or_key_base64_encoded(const char *data)
+{
+    if (NULL == strstr(data, "BEGIN")) {
+        return 0;               // base64 encoded
+    } else {                    // e.g. "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----"
+        return -1;
+    }
+}
+
+static char *kubeconfig_mk_cert_key_tempfile(const char *data)
 {
     static char fname[] = "kubeconfig_mk_tempfile()";
 
-    int decoded_bytes = 0;
-    char *b64decode = base64decode(b64data, strlen(b64data), &decoded_bytes);
-    if (!b64decode || 0 == decoded_bytes) {
-        fprintf(stderr, "%s: Base64 decodes failed.\n", fname);
-        return NULL;
+    const char *cert_key_data = NULL;
+    int cert_key_data_bytes = 0;
+
+    if (0 == is_cert_or_key_base64_encoded(data)) {
+        int decoded_bytes = 0;
+        char *b64decode = base64decode(data, strlen(data), &decoded_bytes);
+        if (!b64decode || 0 == decoded_bytes) {
+            fprintf(stderr, "%s: Base64 decodes failed.\n", fname);
+            return NULL;
+        }
+        cert_key_data = b64decode;
```

Comment:
This memory is leaked, I think. We should free `cert_key_data`

Reply:
Yes. you're right. I will handle it.